File: 프로그래머스/1/12928. 약수의 합/약수의 합.py
def solution(n):
    divisor = []
    answer = 0
    if n == 1:
        return 1
    
    
    for i in range(1, n // 2):
        if n % i == 0:
            divisor.append(i)
            divisor.append(n // i)
    
    divisor = set(divisor)
    answer = sum(divisor)
    
    return answer

File: 프로그래머스/1/12928. 약수의 합/약수의 합.java
class Solution {
    public int solution(int n) {
        int answer = n;
        int limit = n / 2;
        
        for(int i = 1; i <= limit; i++){
            if(n % i == 0){
                answer += i;
            }
        }
        
        return answer;
    }
}